### PR TITLE
CarInfo: Add harness information

### PR DESCRIPTION
**Pull request by sshane**
Notes:
- Fixed and added Niro EV harnesses: 2020: https://github.com/commaai/openpilot/pull/2416, 2022: https://github.com/commaai/openpilot/pull/23253

### CODE REVIEW

File: selfdrive/car/chrysler/values.py [hunk 1]
```diff
@@ -19,16 +21,22 @@ class CAR:
...
   PACIFICA_2020 = "CHRYSLER PACIFICA 2020"
   JEEP_CHEROKEE = "JEEP GRAND CHEROKEE V6 2018"  # includes 2017 Trailhawk
-  JEEP_CHEROKEE_2019 = "JEEP GRAND CHEROKEE 2019" # includes 2020 Trailhawk
+  JEEP_CHEROKEE_2019 = "JEEP GRAND CHEROKEE 2019"  # includes 2020 Trailhawk
+
+
+@dataclass
+class ChryslerCarInfo(CarInfo):
+  package: str = "Adaptive Cruise"
+  harness: str = "fca"
```
Comment (adeebshihadeh) [inline selfdrive/car/chrysler/values.py:30]: let's use enums here and the website can use the same definitions

Comment (sshane) [reply selfdrive/car/chrysler/values.py:30]: A global harness enum in car_definitions?

Comment (sshane) [reply selfdrive/car/chrysler/values.py:30]: In that case, we can remove some extra imports and code to fetch the footnotes and actually define an order if we put the footnotes in a global enum again

Comment (adeebshihadeh) [reply selfdrive/car/chrysler/values.py:30]: yeah, think global enum makes the most sense

**Comment by sshane** [comment]
Some platforms of HKG have different car harnesses by year, do we want to show the same car split across multiple lines with nothing different except year, or automatically detect duplicates and parse the years back to what they were before?

We can also change the structuring a bit and make CarInfo have a list of cars that map year to harness

**Comment by sshane** [comment]
will merge once tests pass

File: selfdrive/car/docs_definitions.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ class CarInfo:
   min_steer_speed: Optional[float] = None
   min_enable_speed: Optional[float] = None
   good_torque: bool = False
+  harness: Optional[Enum] = None
```
Comment (adeebshihadeh) [inline selfdrive/car/docs_definitions.py:53]: don't think you want to set a default here. it shouldn't be easy to forget to specify the harness. defaults make sense for the make subclasses though

Comment (sshane) [reply selfdrive/car/docs_definitions.py:53]: for the makes that specify a default package, we'd need to specify the default harness to also be None, like in Hyundai and Honda, but I can do this. defeats the purpose in those two files

File: selfdrive/car/gm/values.py [hunk 3]
```diff
@@ -62,14 +63,20 @@ class Footnote(Enum):
...
+
+
+CAR_INFO: Dict[str, Union[GMCarInfo, List[GMCarInfo]]] = {
+  CAR.HOLDEN_ASTRA: GMCarInfo("Holden Astra 2017", harness=Harness.custom),
+  CAR.VOLT: GMCarInfo("Chevrolet Volt 2017-18", footnotes=[Footnote.OBD_II], min_enable_speed=0, harness=Harness.custom),
+  CAR.CADILLAC_ATS: GMCarInfo("Cadillac ATS Premium Performance 2018"),
+  CAR.MALIBU: GMCarInfo("Chevrolet Malibu Premier 2017", harness=Harness.custom),
+  CAR.ACADIA: GMCarInfo("GMC Acadia 2018", video_link="https://www.youtube.com/watch?v=0ZN6DdsBUZo", footnotes=[Footnote.OBD_II]),
+  CAR.BUICK_REGAL: GMCarInfo("Buick Regal Essence 2018"),
+  CAR.ESCALADE_ESV: GMCarInfo("Cadillac Escalade ESV 2016", "ACC + LKAS", footnotes=[Footnote.OBD_II]),
```
Comment (sshane) [inline selfdrive/car/gm/values.py:79]: Excluding adding harnesses here as you need two harnesses, so I thought it'd be better if the user read the compatibility page and selected the OBD-II harness and not just see their car and buy without knowing it needs more hardware.

**Pull request merged**